### PR TITLE
Update metadata docstrings

### PR DESCRIPTION
**Pull request by sjvenditto**
## Summary
This PR updates the docstrings for new metadata functions, as well as docstrings for classes that inherit metadata.
- all functions and properties in `metadata_class.py`, notably docstring examples added to `set_info` and `get_info` methods
- `interval_set.py > IntervalSet`, `time_series.py > TsdFrame`, and `ts_group.py > TsGroup`: doc strings that were previously split under class definition and init have been merged, and new docstring examples added for initializing each class with and without metadata
- docstrings were added to some properties throughout `IntervalSet`, `TsGroup`, `TsdFrame`, `_BaseTsd`, and `_Base`
- "Metadata" section on the "User Guide > Core methods" documentation page has been updated

### CODE REVIEW

File: doc/user_guide/04_core_methods.md [hunk 1]
```diff
@@ -373,3 +267,256 @@ plt.xlabel("Time (s)")
...
+```
+
+### Metadata properties
+User-set metadata is mutable and can be overwritten.
+```{code-cell} ipython3
+tsgroup.set_info(region=["A","B","C"])
+print(tsgroup.get_info("region"))
+```
+
+If the metadata name does not overlap with an existing class column, it can be set and accessed via key indexing (i.e. using square brackets).
```
Comment (gviejo) [inline doc/user_guide/04_core_methods.md:494]: Shouldn't this be in the section Accessing metadata?

Comment (gviejo) [reply doc/user_guide/04_core_methods.md:494]: Also why mentioning class column only?

File: pynapple/core/metadata_class.py
Comment (gviejo) [inline pynapple/core/metadata_class.py:1]: Ok something very annoying but for the docstrings of set_info, the examples appears as TsGroup for IntervalSet. Is there a way to have dynamical examples as a function of what is inherited?

File: pynapple/core/ts_group.py
Comment (gviejo) [inline pynapple/core/ts_group.py:1]: Ok here it's different from IntervalSet and TsdFrame. The init is basically empty of docstrings now. I think I prefer this way for the documentation.

File: pynapple/core/metadata_class.py
Comment (sjvenditto) [reply pynapple/core/metadata_class.py:1]: This is a good idea, I'll look into it

File: pynapple/core/ts_group.py
Comment (sjvenditto) [reply pynapple/core/ts_group.py:1]: I think this comment in on an older version of `ts_group.py`, it should also have an empty init

File: pynapple/core/metadata_class.py
Comment (sjvenditto) [reply pynapple/core/metadata_class.py:1]: added class-specific examples

File: pynapple/core/ts_group.py
Comment (sjvenditto) [reply pynapple/core/ts_group.py:1]: removed init strings from intervalset and tsdframe

File: doc/user_guide/04_core_methods.md (hunk 1 above)
Comment (sjvenditto) [reply doc/user_guide/04_core_methods.md]: I changed this to explicitly mention the exceptions for IntervalSet's "start" and "end"

**Review by gviejo — APPROVED** [review]

**Pull request merged**